src/file_lines.rs

```diff
@@ -151,6 +151,21 @@ impl<'a> iter::Iterator for Files<'a> {
     }
 }
 
+fn canonicalize_path_string(s: &str) -> String {
+    use std::path;
+    use std::fs;
+
+    match fs::canonicalize(path::PathBuf::from(s)) {
+        Ok(canonicalized) => {
+            match canonicalized.to_str() {
+                Some(c) => c.to_string(),
+                _ => String::new(),
+            }
+        }
+        _ => String::new(),
```

Comment:
Yes, I wasn't 100% happy with that bit. I've changed `canonicalize_path_string` to return an `Option<String>` in the latest commit, then integrated that with the error handling in the calling functions.